Sources/WebSockets/Client/WebSocket+Client.swift

```diff
@@ -50,6 +54,7 @@ extension WebSocket {
         )
         let response = try client.respond(to: request)
 
+        guard response.status == .switchingProtocols else { throw ResponseError.status(response.status) }
```

Comment:
In my case, I am using a Middleware test that is responding with 'unauthorized'.  I am not sure that 'invalidOrUnsupportedStatus' as an error code is as expressive as wrapping the actual status code and providing in the Throw.    I also don't think that it falls under a 'FormatError' classification.

I put it in front of the other tests, since those tests likely only apply if the response is 'switchingProtocols'.  

Can I ask you to reconsider?

Can I suggest replacing the functionality of L61 with my code?